iOS-training/LayoutPrototype.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  LayoutPrototype.swift
+//  iOS-training
+//
+//  Created by 松本 幸太郎 on 2023/07/10.
+//
+
+import SwiftUI
+import YumemiWeather
+
+struct LayoutPrototype: View {
+    @State var fetchedWeather: Weather? = nil
+
+    var body: some View {
+        GeometryReader { geometry in
+            VStack(alignment: .center, spacing: .zero) {
+                if fetchedWeather != nil {
+                    fetchedWeather!.icon
+                        .resizable()
+                        .scaledToFit()
+                        .foregroundStyle(fetchedWeather!.color)
+                        .frame(width: geometry.size.width / 2,
+                               height: geometry.size.width / 2)
+                }
```

Comment:
[must]
```suggestion
                if let fetchedWeather {
                    fetchedWeather.icon
                        .resizable()
                        .scaledToFit()
                        .foregroundStyle(fetchedWeather.color)
                        .frame(width: geometry.size.width / 2,
                               height: geometry.size.width / 2)
                }
```
基本的にforce unwrappingは避けましょう。nilでない場合にその値を使って何か処理を行う場合は`if let` や `guard let`といった記法を使うと良いです。
ちなみに、今回の`if let  fetchedWeather {` は`if let  fetchedWeather = fetchedWeather {`の省略形です。

Reply:
```if let  fetchedWeather {```という書き方を初めて知りました。
ありがとうございます！

Reply:
The Swift Programming Language は内容が充実しているので、一度目を通してもらうと良いと思いました。

https://docs.swift.org/swift-book/documentation/the-swift-programming-language/thebasics/#Optional-Binding